File: Java/src/main/java/com/di/di/services/ComponentBService.java
package com.di.di.services;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class ComponentBService {
    private Logger logger;
    private SingletonState singletonState;
    private TransientState transientState;

    @Autowired
    public ComponentBService(Logger logger, SingletonState singletonState, TransientState transientState) {
        this.logger = logger;
        this.singletonState = singletonState;
        this.transientState = transientState;
    }

    public void DoIt(){
        logger.LogInformation(String.format("ComponentBService (SingletonState) %s", singletonState.GetValue()));
        logger.LogInformation(String.format("ComponentBService (TransientState) %s", transientState.GetValue()));
    }
}

File: CPlusPlus/services/ComponentBService.cpp
#ifndef COMPONENTBSERVICE
#define COMPONENTBSERVICE

#include "..\di.hpp"
#include "Logger.cpp"
#include "SingletonState.cpp"
#include "TransientState.cpp"

class ComponentBService {
    Logger* _logger;
    SingletonState* _singletonState;
    TransientState _transientState;

    public:
        BOOST_DI_INJECT(ComponentBService, 
            std::shared_ptr<Logger> logger, 
            std::shared_ptr<SingletonState> singletonState, 
            TransientState transientState) {
            _logger = logger.get();
            _singletonState = singletonState.get();
            _transientState = transientState;
        }

        void DoIt() {
            _logger->LogInfo("ComponentBService (SingletonState) " + std::to_string(_singletonState->GetValue()));
            _logger->LogInfo("ComponentBService (TransientState) " + std::to_string(_transientState.GetValue()));
        }
};

#endif